File: Greedy_Algorithm/로프/2217.py
n = int(input())
arr = []
for i in range(n):
    temp = int(input())
    arr.append(temp)

arr.sort()

max = arr[0] * n
for i in range(n):
    if max < arr[i] * (n-i):
        max = arr[i] * (n-i)

print(max)

File: Greedy_Algorithm/로프/2217.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(){
    int n;  cin >> n;
    vector<int> arr(n,0);
    for(int i=0; i<n; i++)
        cin >> arr[i];

    sort(arr.begin(), arr.end());

    int max = arr[0] * n;
    for(int i=0; i<n; i++)
        if(max < arr[i] * (n - i))
            max = arr[i] * (n - i);
    cout << max;
}